synapse/handlers/room_member.py

```diff
@@ -183,6 +185,44 @@ def update_membership(
             third_party_signed=None,
             ratelimit=True,
     ):
+        if action == Membership.JOIN:
+            result = self.join_cache.get((room_id, target))
```

Comment:
Should ``remote_room_hosts`` and ``third_party_signed`` be part of the request key here?